contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,860 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @title Aerodrome AMO strategy
+ * @author Origin Protocol Inc
+ */
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+import { IERC20, InitializableAbstractStrategy } from "../../utils/InitializableAbstractStrategy.sol";
+import { StableMath } from "../../utils/StableMath.sol";
+
+import { ISugarHelper } from "../../interfaces/aerodrome/ISugarHelper.sol";
+import { INonfungiblePositionManager } from "../../interfaces/aerodrome/INonfungiblePositionManager.sol";
+import { ISwapRouter } from "../../interfaces/aerodrome/ISwapRouter.sol";
+import { ICLPool } from "../../interfaces/aerodrome/ICLPool.sol";
+import { ICLGauge } from "../../interfaces/aerodrome/ICLGauge.sol";
+import { IVault } from "../../interfaces/IVault.sol";
+
+import "hardhat/console.sol";
+
+contract AerodromeAMOStrategy is InitializableAbstractStrategy {
+    using StableMath for uint256;
+    using SafeERC20 for IERC20;
+    using SafeCast for uint256;
+
+    /***************************************
+            Storage slot members
+    ****************************************/
+
+    /// @notice tokenId of the liquidity position
+    uint256 public tokenId;
+    /// @dev Minimum amount of tokens the strategy would be able to withdraw from the pool.
+    ///      minimum amount of tokens are withdrawn at a 1:1 price
+    uint256 public underlyingAssets;
+    /// @notice the gauge for the corresponding Slipstream pool (clPool)
+    /// @dev can become an immutable once the gauge is created on the base main-net
```

Comment:
Can't this be immutable now, since having a gauge in fork tests doesn't depend on already having the strategy deployed? We can create the gauge, then pass it to the constructor.

Reply:
Great point thanks. Done here: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/876a4e4d53d580097383c168261f0db2555c6259